lib/libdyn/src/dynamic.c

```diff
@@ -0,0 +1,22 @@
+#include "dynamic.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+double mypow_dynamic(double base, int exp)
+{
+    if (exp == 0)
+        return 1.0;
+
+    double result = base;
+
+    // multiplico exp-1 veces la base
+    for (int i = 0; i < abs(exp) - 1; ++i)
+        result *= base;
+
+    if (exp < 0)
+        result = 1 / result;
```

Comment:
I think it is better to use the provided C math library, and not reinvent the wheel